src/identifiers.md

```diff
@@ -2,10 +2,34 @@
 
 > **<sup>Lexer:<sup>**  
 > IDENTIFIER :  
-> &nbsp;&nbsp; &nbsp;&nbsp; XID_start XID_continue<sup>\*</sup>  
-> &nbsp;&nbsp; | `_` XID_continue<sup>+</sup>  
+> &nbsp;&nbsp; &nbsp;&nbsp; [`a`-`z` `A`-`Z`]&nbsp;[`a`-`z` `A`-`Z` `0`-`9` `_`]<sup>\*</sup>  
+> &nbsp;&nbsp; | `_` [`a`-`z` `A`-`Z` `0`-`9` `_`]<sup>+</sup>  
 
-An identifier is any nonempty Unicode[^non_ascii_idents] string of the following form:
+An identifier is any nonempty ASCII[^non_ascii_idents] string of the following form:
+
+Either
+
+   * The first character is a letter
+   * The remaining characters are alphanumeric or `_`
+
+Or
+
+   * The first character is `_`
+   * The identifier is more than one character, `_` alone is not an identifier
+   * The remaining characters are alphanumeric or `_`
+
+[keywords]: keywords.html
+
+[^non_ascii_idents] Non-ASCII characters in identifiers are currently feature-gated. See [issue #28979](https://github.com/rust-lang/rust/issues/28979).
```

Comment:
I'd snip everything below this. You can move the comment into the Unstable Book, or it's in the git history. This also involves removing the link from the text above.